cirq-core/cirq/sim/simulation_state.py

```diff
@@ -298,9 +307,14 @@ def strat_act_on_from_apply_decompose(
     qubit_map = {q: qubits[i] for i, q in enumerate(qubits1)}
     if operations is None:
         return NotImplemented
+    ancillas = list(set(q for op in operations for q in op.qubits if q not in qubits1))
```

Comment:
Just a comment on the early return value above while you're here: that should be moved two lines up, immediately after decomposing into the operations.